zerver/webhooks/semaphore2/view.py

```diff
@@ -0,0 +1,180 @@
+    author_email = payload["revision"]["sender"]["email"]
+    branch_name = payload["revision"]["branch"]["name"]
+    blocks = get_blocks_data(payload["blocks"])
+
+    if payload["revision"]["pull_request"] is None:
+        event = "push"
+    else:
+        event = "pull_request"
+        pull_request_title = payload["revision"]["pull_request"]["name"]
+        pull_request_number = payload["revision"]["pull_request"]["number"]
+        pull_request_url = get_pr_url(pull_request_number, repository_url)
+
+    if event == "push":
+        commit_id = payload["revision"]["commit_sha"]
+        commit_message = payload["revision"]["commit_message"]
+        commit_url = get_commit_url(commit_id, repository_url)
+        content = PUSH_TEMPLATE.format(
+            email = author_email,
+            branch_name = branch_name,
+            organization_name = organization_name,
+            project_name = project_name,
+            repository_url = repository_url,
+            commit_hash = commit_id[:7],
+            commit_message = commit_message,
+            commit_url = commit_url,
+            pipeline_name = pipeline_name,
+            pipeline_result = pipeline_result,
+            blocks = blocks,
+        )
+
+    elif event == "pull_request":
+        content = PULL_REQUEST_TEMPLATE.format(
+            email = author_email,
+            branch_name = branch_name,
+            organization_name = organization_name,
+            project_name = project_name,
+            repository_url = repository_url,
+            pull_request_title = pull_request_title,
+            pull_request_url = pull_request_url,
+            pipeline_name = pipeline_name,
+            pipeline_result = pipeline_result,
+            blocks = blocks,
+        )
+
+    else:  # nocoverage
+        content = DEFAULT_TEMPLATE.format(
+            email = author_email,
+            branch_name = branch_name,
+            organization_name = organization_name,
+            project_name = project_name,
+            repository_url = repository_url,
+            pipeline_name = pipeline_name,
+            pipeline_result = pipeline_result,
+            blocks = blocks,
+        )
+
+    subject = TOPIC_TEMPLATE.format(
+        project=project_name,
+        branch=branch_name
+    )
+
+    check_send_webhook_message(request, user_profile, subject, content)
+    return json_success()
+
+#Helper Functions
+def get_blocks_data(blocks: List[Dict[str, Any]]) -> str:
+    """
+        Returns the markdown presentation of data related to blocks.
+    """
+    BLOCKS_DATA = """"""
+    BLOCK_BODY = """
+**1. {name}**:
+  - Result: {result}
+""".strip()
+    for block in blocks:
+        name = block["name"].title()
+        result = block["result"]
+        content = BLOCK_BODY.format(name=name, result=result)
+        BLOCKS_DATA += content
+
+    return BLOCKS_DATA
+
+# Funtions to generate url to a commit/Pull Request. This builds an url
```

Comment:
This comment is not necessary, I think. The function names and doc strings below are good enough.

Reply:
Okay, I'll remove that comment.